doc/mkdocs/data/create_new_library.md

```diff
@@ -0,0 +1,644 @@
+
+#### Block Mask
+
+A Simulink Block is just a square with input and output ports. It is possible to "mask" a Simulink Block in order to provide a user-friendly GUI that can be useful e.g. for setting Block parameters. The mask may contain buttons, dropdown menus, ...
+
+#### Shared library
+
+A shared library is an object that contains compiled code (functions, classes, ...) which cannot be executed standalone. They can be either _static_ or _dynamic_. Static shared libraries are meant to be embedded inside the executable that calls their symbols, instead the code of dynamic shared libraries is only referenced inside the executable and called when needed.
+
+For grasping better this difference, if multiple executables link against the same static library, the same code is embedded inside all of them, resulting in bigger executables and code duplication. A dynamic library object instead can be shared by multiple executables that need only to know their location in the filesystem and which symbols they provide.
+
+#### Plugin Library
+
+A plugin library is a particular type of a dynamic shared library. An executable can load it either at load time, i.e. at the beginning, or at run time, i.e. when needed during the execution. The libraries that can be loaded during runtime are called plugins.
+
+!!! info
+    On UNIX systems the load the a plugin is executed by a `dlopen`
+
+### Overview of Simulink execution
+
+The core of the WB-Toolbox provides a framework capable of loading during runtime shared libraries. When the Simulink simulation starts, its engine assigns a deterministic order to the blocks of the Simulink Model. If one of these blocks is not contained in the system toolboxes, it needs to be associated to a _S-Function_ that implements its logic. The toolbox provides a streamlined approach to implement these functions without the need of being an expert of the complex Simulink APIs.
+
+The blocks of our library are implemented in Simulink using the following _S-Function_ block:
+
+![EmptySFunctionBlock](images/EmptySFunctionBlock.png)
+
+What you need to know for the time being is that the name of the plugin library generated from the C++ code will be stored in the _S-function parameters_ field. We provide a generic **Level-2 MEX S-Function** that reads this parameter and uses the library name name to dynamically load the dynamic shared library that contains the block's code.
+
+After the library is found in the filesystem and successfully loaded, Simulink allocates an object of the C++ class associated to the block functionality. Again, this information (the class name) is passed in the _S-function parameters_ field.
+
+## Develop the C++ plugin
+
+You already learned that Blocks are no more than regular C++ classes. They are not an exception, in fact all the components discussed until now are mapped to C++ classes or interfaces (virtual classes).
```

Comment:
interface  (virtual classes) ---> interfaces (pure virtual classes)

Reply:
Or  perhabs https://en.cppreference.com/w/cpp/language/abstract_class is better name?

Reply:
> interface (virtual classes) ---> interfaces (pure virtual classes)

I didn't get what you meant here, but I agree that using abstract class would be better.

Reply:
Neither the C++ standard nor the C++ book by Bjarne Stroustrup define the "virtual class" term. However, as far as I checked on the internet (see for example https://en.wikipedia.org/wiki/Virtual_class), "virtual classes" typical refer to any class some methods of which can be override.  Classes in which some method implementation is missing at all (as typically done in interfaces) are sometime called **pure** virtual classes.